doc/developer-notes.md

```diff
@@ -39,6 +39,7 @@ code.
   - `++i` is preferred over `i++`.
   - `nullptr` is preferred over `NULL` or `(void*)0`.
   - `static_assert` is preferred over `assert` where possible. Generally; compile-time checking is preferred over run-time checking.
+  - `enum class` is preferred over `enum` where possible. Scoped enumerations avoid three potential pitfalls/problems with traditional C++ enumerations: 1.) implicit conversions to int, 2.) name clashes due to enumerators being exported to the surrounding scope, and 3.) inability to specify the underlying type of the enum.
```

Comment:
I had the impression it was possible to specify the type even for the "deprecated" enum.

Reply:
@MarcoFalke Ah, you're right. Forward declaration of `enum`:s was fixed in C++11. Will update!

Reply:
It's ok, avoiding scope leak is the best argument for using scoped enums anyway.